container/lru/set_test.go

```diff
@@ -0,0 +1,671 @@
+func putSetTTLAndAssertNumEvicted(tc *testContextSet, item uint64, ttl time.Duration, wantNumEvicted uint32) {
+	tc.t.Helper()
+
+	gotNumEvicted := tc.s.PutWithTTL(item, ttl)
+	if gotNumEvicted != wantNumEvicted {
+		tc.t.Fatalf("%q: PutWithTTL returned unexpected number of evicted "+
+			"items: got %v, want %v", tc.name, gotNumEvicted, wantNumEvicted)
+	}
+}
+
+// assertSetItems asserts the set associated with the provided test context
+// returns the provided items from Items.
+func assertSetItems(tc *testContextSet, wantItems []uint64) {
+	tc.t.Helper()
+
+	gotItems := tc.s.Items()
+	if !reflect.DeepEqual(gotItems, wantItems) {
+		tc.t.Fatalf("%q: unexpected items: got %v, want %v", tc.name, gotItems,
+			wantItems)
+	}
+}
+
+// assertSetMembership asserts the set associated with the provided test context
+// reports the given expected existence for Exists and Contains.
+func assertSetMembership(tc *testContextSet, item uint64, shouldExist bool) {
+	tc.t.Helper()
+
+	ok := tc.s.Exists(item)
+	if ok != shouldExist {
+		tc.t.Fatalf("%q: Exists item %d: got %v, want %v", tc.name, item, ok,
+			shouldExist)
+	}
+
+	ok = tc.s.Contains(item)
+	if ok != shouldExist {
+		tc.t.Fatalf("%q: Contains item %d: got %v, want %v", tc.name, item, ok,
+			shouldExist)
+	}
+}
+
+// assertInternSetMembership asserts the set associated with the provided test
+// context has an internal state that matches the given expected existence.
+func assertInternSetMembership(tc *testContextSet, item uint64, shouldExist bool) {
+	tc.t.Helper()
+
+	tc.s.m.mtx.Lock()
+	_, exists := tc.s.m.items[item]
+	tc.s.m.mtx.Unlock()
+
+	if exists != shouldExist {
+		tc.t.Fatalf("%q: unexpected physical existence for item %d: got %v, "+
+			"want %v", tc.name, item, exists, shouldExist)
+	}
+}
+
+// evictSetExpiredAndAssertNumEvicted asserts that calling EvictExpiredNow on
+// the set associated with the provided test context returns the given expected
+// number of evicted items.
+func evictSetExpiredAndAssertNumEvicted(tc *testContextSet, wantNumEvicted uint32, expirationDisabled bool) {
+	tc.t.Helper()
+
+	gotNumEvicted := tc.s.EvictExpiredNow()
+	if expirationDisabled {
+		wantNumEvicted = 0
+	}
+	if gotNumEvicted != wantNumEvicted {
+		tc.t.Fatalf("%q: EvictExpiredNow returned unexpected number of "+
+			"evicted items: got %v, want %v", tc.name, gotNumEvicted,
+			wantNumEvicted)
+	}
+}
+
+// assertSetHitRatio asserts the set associated with the provided test context
+// returns the expected hit ratio.
+func assertSetHitRatio(tc *testContextSet, wantRatio float64) {
+	tc.t.Helper()
+
+	gotRatio := tc.s.HitRatio()
+	roundedRatio := math.Round(gotRatio*100) / 100
+	if roundedRatio != wantRatio {
+		tc.t.Fatalf("%q: unexpected hit ratio: got %v, want %v", tc.name,
+			roundedRatio, wantRatio)
+	}
+}
+
+// TestSet ensures the set behaves as expected including limiting, eviction of
+// least recently used entries, specific entry removal, peeking, existence and
```

Comment:
Agreed that it's not very clear.

Updated to call out existence and containment  (since `Exists` is the peek I was referring to there with sets while `Contains` is the containment check that also exhibits the typical LRU behavior).